src/regressionmodel.jl

```diff
@@ -132,3 +132,43 @@ function linearpredictor end
 In-place version of [`linearpredictor`](@ref), storing the result in `storage`.
 """
 function linearpredictor! end
+
+"""
+    vif(m::RegressionModel)
+
+Compute the variance inflation factor (VIF).
+
+Returns a vector of inflation factors computed for each coefficient except
+for the intercept.
+In other words, the corresponding coefficients are `coefnames(m)[2:end]`.
+
+The [variance inflation factor (VIF)](https://en.wikipedia.org/wiki/Variance_inflation_factor) measures
+the increase in the variance of a parameter's estimate in a model with multiple parameters relative to
+the variance of a parameter's estimate in a model containing only that parameter.
+
+See also [`coefnames`](@ref), [`gvif`](@ref).
+
+!!! warning
+    This method will fail if there is (numerically) perfect multicollinearity,
+    i.e. rank deficiency. In that case though, the VIF
+    is not particularly informative anyway.
+"""
+function vif(model::RegressionModel)
+    mm = Statistics.cov2cor!(vcov(model), stderror(model))
+    # TODO: replace with hasintercept() when implemented (xref #17)
+    all(==(1), view(modelmatrix(model), :, 1)) ||
+        throw(ArgumentError("VIF only defined for models with an intercept term"))
+    mm = @view mm[2:end, 2:end]
```

Comment:
```suggestion
    i = findfirst(Base.Fix1(all, isone), eachcol(modelmatrix(X)))
    i === nothing &&
        throw(ArgumentError("VIF is only defined for models with an intercept term"))
    # Translate the column index in the model matrix to the corresponding indices in the
    # correlation matrix
    j1 = firstindex(mm, 1) - i + 1
    j2 = firstindex(mm, 2) - i + 1
    m = view(mm, axes(mm, 1) .!= j1, axes(mm, 2) .!= j2)
```
The intercept term doesn't necessarily have to be the first column in the model matrix. We also can't assume that `modelmatrix(model)` has 1-based indexing nor that it shares its indexing behavior with the covariance matrix. Also note that in this suggestion I've changed the variable name of the view from `mm` to `m` so that each variable's type remains constant. (Currently `cov2cor!` requires 1-based indexing but that may not always be true; there may also be packages that provide additional methods for that function which don't require 1-based indexing.)